src/yaml-languange-support/schema-utils.ts

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * An utility function for splitting text into lines.
+ */
+export function splitLines(text) {
+    return text.split(/\r\n?|\n/);
+} 
+
+/**
+ * A very easy utility function for decides whether the text is support to be a kubernetes yaml file, further
+ * implementations is needed to be more precise on the result rather than the regex test.
+ */
+export function isKubernetesYamlFile(text) {
+    return !!(/^kind\s*\:/g.exec(text) ||
+        /[^a-zA-Z]kind\s*\:/g.exec(text) ||
+        /^apiVersion\s*\:/g.exec(text) ||
+        /[^a-zA-Z]apiVersion\s*\:/g.exec(text)
+    );
+}
+
+/**
+ * An utility function for get apiVersion and kind from 'x-kubernetes-group-version-kind' node in kubernetes schema.
+ */
+export function parseApiVersionKind(groupKindNode) {
+    if (!groupKindNode.version) {
+        throw new Error("Invalid 'x-kubernetes-group-version-kind' node, 'version' is missing.");
```

Comment:
What are we planning to do with these errors?  The messages are extremely technical - few users will know what an `x-kubernetes-group-version-kind` is - so we can't display them...